src/components/flowchart/draw.js

```diff
@@ -171,6 +172,10 @@ export const drawNodes = function (changed) {
       )
       .classed('pipeline-node--data', (node) => node.type === 'data')
       .classed('pipeline-node--task', (node) => node.type === 'task')
+      .classed(
+        'pipeline-node--memory-data',
+        (node) => flags?.diffMemoryDatasets && node?.dataset?.includes('memory')
```

Comment:
This looks good. Can we have a DatasetType check instead of string matching ?

Reply:
yes, i think we will eventually refactor and do it properly. for now this is just a quick dirty experiment